### PR TITLE
virtualbox: 6.1.30 -> 6.1.34

### PR DESCRIPTION
**Pull request by andresilva**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/virtualbox/ffreestanding.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ TEMPLATE_VBoxR0_TOOL                = $(VBOX_GCC_TOOL)
+ TEMPLATE_VBoxR0_CFLAGS              = -fno-pie -nostdinc -g $(VBOX_GCC_pipe) $(VBOX_GCC_WERR) $(VBOX_GCC_PEDANTIC_C) \
+ 	$(VBOX_GCC_Wno-variadic-macros) $(VBOX_GCC_R0_OPT) $(VBOX_GCC_R0_FP) -fno-strict-aliasing -fno-exceptions \
+-	$(VBOX_GCC_fno-stack-protector) -fno-common $(VBOX_GCC_fvisibility-hidden) -std=gnu99 $(VBOX_GCC_IPRT_FMT_CHECK)
++	$(VBOX_GCC_fno-stack-protector) -fno-common -ffreestanding $(VBOX_GCC_fvisibility-hidden) -std=gnu99 $(VBOX_GCC_IPRT_FMT_CHECK)
+ TEMPLATE_VBoxR0_CXXFLAGS            = -fno-pie -nostdinc -g $(VBOX_GCC_pipe) $(VBOX_GCC_WERR) $(VBOX_GCC_PEDANTIC_CXX) \
+ 	$(VBOX_GCC_Wno-variadic-macros) $(VBOX_GCC_R0_OPT) $(VBOX_GCC_R0_FP) -fno-strict-aliasing -fno-exceptions \
+ 	$(VBOX_GCC_fno-stack-protector) -fno-common $(VBOX_GCC_fvisibility-inlines-hidden) $(VBOX_GCC_fvisibility-hidden) \
+ 	-fno-rtti $(VBOX_GCC_IPRT_FMT_CHECK)
++ if $(VBOX_GCC_VERSION_CC) >= 40500 # 4.1.2 complains, 4.5.2 is okay, didn't check which version inbetween made it okay with g++.
```
Comment (toastal) [inline pkgs/applications/virtualization/virtualbox/ffreestanding.patch:15]: inbetween -> in between

**Review by toastal — CHANGES_REQUESTED** [review]

File: pkgs/applications/virtualization/virtualbox/ffreestanding.patch (hunk 1 above)
Comment (andresilva) [reply pkgs/applications/virtualization/virtualbox/ffreestanding.patch:15]: The patches were taken as is from upstream, I'd prefer to keep them unchanged.

Comment (toastal) [reply pkgs/applications/virtualization/virtualbox/ffreestanding.patch:15]: That is fair.

**Comment by toastal** [comment]
@andresilva My `nix log` trying to use this merge request https://paste.sr.ht/~toastal/a053035fb4116d5e170b3f48a20807d12c2f47eb

**Comment by SuperSandro2000** [comment]
relevant lines

```
[01m[K../drivers/i2c/busses/i2c-nct6775.c:[m[K In function '[01m[Knct6775_access[m[K':
[01m[K../drivers/i2c/busses/i2c-nct6775.c:221:39:[m[K [01;31m[Kerror: [m[Kthis statement may fall through [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-fallthrough=-Werror=implicit-fallthrough=]8;;[m[K]
  221 |                         [01;31m[Ktmp_data.byte = data->byte[m[K;
      |                         [01;31m[K~~~~~~~~~~~~~~^~~~~~~~~~~~[m[K
[01m[K../drivers/i2c/busses/i2c-nct6775.c:222:17:[m[K [01;36m[Knote: [m[Khere
  222 |                 [01;36m[Kcase[m[K I2C_SMBUS_BYTE:
      |                 [01;36m[K^~~~[m[K
  CC [M]  drivers/usb/cdns3/cdns3-plat.o
cc1: all warnings being treated as errors
make[3]: *** [../scripts/Makefile.build:288: drivers/i2c/busses/i2c-nct6775.o] Error 1
make[2]: *** [../scripts/Makefile.build:550: drivers/i2c/busses] Error 2
make[1]: *** [../scripts/Makefile.build:550: drivers/i2c] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by andresilva** [comment]
That log seems to be the compilation of a linux kernel and not really virtualbox, unless I misunderstood something.

**Comment by toastal** [comment]
That's likely... I am on Zen, but I've not yet ran into that specific issue because I've been blocked by VirtualBox flagged as broken :sweat_smile:

**Pull request merged**